enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,219 @@
+cluster controls if/how it is implemented in the HCP.
+
+`HyperShiftLogForwarder` is a subset of the `ClusterLogForwarder` API:
+- Only the `audit` input is allowed, and only API server audit logs will be forwarded.
+- There may be restrictions on the set of outputs supported from the management cluster.
+  - Possibly only a predefined set of named outputs will be allowed.
+- There may be restrictions on pipeline features that are not relevant to API audit logs.
+
+### Implementation Details
+
+#### ClusterLogForwarder multiple instances
+
+The ClusterLogForwarder is currently a singleton.
+To provide isolation between HCPs we need to deploy a forwarder per HCP namespace.
+This requires modifications to the CLO 
+
+_FIXME JIRA ref._
+
+#### ClusterLogForwarder HTTP listener
+
+The ClusterLogForwarder needs to be extended to act a a HTTP listener,
+so it can receive logs as a web hook from the API server in the HCP.
+
+_FIXME JIRA reference_
+
+_FIXME Is there a way to use the file system and avoid this requirement?_
+
+#### Hypershift Logging Operator
+
+This operator must
+- watch resources in a HCP API server, update their status 
+  - `HypershiftLogForwarder` 
+  - `Policy.audit.k8s.io`
+- create resources in the management cluster API server
+  - `ClusterLogForwarder`
+  - `Policy.audit.k8s.io`
+
+The policy objects can be copied verbatim. 
+
+The `HypershiftLogForwarder` fields are compatible with `ClusterLogForwarder` fields but some changes
+and validation may be needed:
+- validate that the forwarding request is valid and allowed.
+- possibly replace named outputs with a suitable output configuration;
+  if we want to restrict the user to a fixed set of alternatives
+- possibly add credentials derived from the guest cluster or the management cluster.
+  
+### Risks and Mitigations
+
+CPU/Network cost of extra log forwarding
+- We have to do this anyway, but user might choose a more verbose policy.
+- We could restrict the available logs at the API server to control this.
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions
+
+- Is there any way to write the audit logs to a file system that the log forwarder can read?
+  This would be simpler & more reliable than a web hook.
+- Can an operator on the management cluster reconcile and update status on resources in the guest cluster directly?
+  The API server is running in the HCP on the management cluster, but need to check for possible snags.
+- Should we allow the user to specify any destination for logs from the management cluster? If not:
+   - Why not? We allow that freedom on self contained clusters. I see no new security/privacy issues.
+   - We can instead provide a fixed set of named alternatives. The `HyperShiftLoggingOperator`
+     can translate those names into known configurations.
+- How do we manage credentials? Assuming they are stored in the HCP API server, can the management cluster
+  access secrets for TLS security and the like? If not how do we get at them.
```

Comment:
Recieving events from the HCP APIs will be via webhook specific to that HCP.  After that is processed it's up to how we architect this solution with respect to how CLO receives the events.

The challenge is when we look at authentication to various destinations.  The solution needs to support things like CloudWatch for ROSA HCP.  This requires STS, which requires a token per HCP.  Tokens expire after 60 minutes.  CLO will need to refresh tokens from each HCP.  We have code for this in the current solution (on gitlab) that can be referenced.

Reply:
+1. The CLF resource references all the relevant credentials and connection info, the HLF resource would do the same. See other proposal: The HLO on the MC would use the HLF from the hosed cluster as a template to construct a CLF in the HCP that does the actual forwarding.